src/test/scala/chiselTests/experimental/hierarchy/SeparateElaborationSpec.scala

```diff
@@ -0,0 +1,119 @@
+      )
+
+      // Grab DUT definition to pass into testbench
+      val designDefs = dutAnnos.flatMap { a =>
+        a match {
+          case a: DesignAnnotation[_] =>
+            Some(a.design.asInstanceOf[AddOne].toDefinition)
+          case _ => None
+        }
+      }
+      require(designDefs.length == 1, s"Exactly one DesignAnnotation should exist, but found: $designDefs.")
+      val dutDef = designDefs.head
+
+      class Testbench(defn: Definition[AddOne]) extends Module {
+        // Make sure names do not conflict
+        val mod = Module(new AddOne)
+        val inst = Instance(defn)
+
+        // Tie inputs to a value so ChiselStage does not complain
+        mod.in := 0.U
+        inst.in := 0.U
+      }
+
+      (new ChiselStage).run(
+        Seq(
+          ChiselGeneratorAnnotation(() => new Testbench(dutDef)),
+          TargetDirAnnotation(testDir),
+          ImportedDefinitionAnnotation(dutDef)
+        )
+      )
+
+      // Check that the output RTL has only a module instantiation and no
+      // module declaration.
+      val tb_rtl = Source.fromFile(s"$testDir/Testbench.v").getLines.mkString
+      tb_rtl should include("AddOne inst (")
+      (tb_rtl should not).include("module AddOne")
+    }
+  }
+
+  describe("(1): Elaborating an Instance and Definition together") {
+    it("should not result in a repeat definition of the module.") {
+      val testDir = createTestDirectory(this.getClass.getSimpleName).toString
+
+      class Top extends Module {
+        val inst = Instance(Definition(new AddOne))
+        inst.in := 0.U
+      }
+
+      // If there is a repeat module definition, FIRRTL emission will fail
+      (new ChiselStage).emitFirrtl(
+        gen = new Top,
+        args = Array("-td", testDir)
+      )
+    }
+  }
+
+  describe("(2): Elaborating multiple Instances separately from its Definition") {
+    it("should not result in a repeat definition of the module.") {
+      val testDir = createTestDirectory(this.getClass.getSimpleName).toString
+      val dutAnnos = (new ChiselStage).run(
+        Seq(
+          ChiselGeneratorAnnotation(() => new AddOne),
+          TargetDirAnnotation(testDir)
+        )
+      )
+
+      // Grab DUT definition to pass into testbench
+      val designAnnos = dutAnnos.flatMap { a =>
+        a match {
+          case a: DesignAnnotation[_] =>
+            Some(a.design.asInstanceOf[AddOne].toDefinition)
+          case _ => None
+        }
+      }
+      require(designAnnos.length == 1, s"Exactly one DesignAnnotation should exist, but found: $designAnnos.")
+      val dutDef = designAnnos.head
+
+      class Testbench(defn: Definition[AddOne]) extends Module {
+        val inst0 = Instance(defn)
+        val inst1 = Instance(defn)
+
+        inst0.in := 0.U
+        inst1.in := 0.U
+      }
+
+      // If there is a repeat module definition, FIRRTL emission will fail
+      val firrtl = (new ChiselStage).emitFirrtl(
+        gen = new Testbench(dutDef),
```

Comment:
Right now, the only way to instantiate a definition from a previous elaboration is to pass it as an argument to the top module in the next elaboration. This means that we must support a better API (e.g. like Instantiate), prior to release.